### DISCUSSION TITLE
[7] Dinder - ASR

### DISCUSSION THREAD

**Discussion by Manuel-F-04** [dhbw-ka-tinf22b5-dinder/Dinder#12] [Projektblog]
### Utility Tree
| Q-Attribute |	Refinement |	Quality attribute scenarios |	Business Value |	Technical Risk |
|-------------|------------|------------------------------|----------------|-----------------|
| Interoperability |	API-Specification |	on localhost:8080/api you are able to view endpoint specifications |	M |	L |
| Modifiable |	granular-structured backend |	Through the standardized use of spring boot and a relational database schema is the backend easily expandable | H |	H |			
|Useability	| User-Feedback	| The user can tell us how can improve even further	| M	| M |
| |Intuitive Design |	The website guides the user through its intuitive and simple design	| H |	M |
|Testability	| Testing frameworks	| During the development phase we use automated tests to help fulfill our quality requirements	| H	| M |

### Architectural decisions:
- We want to write tests in the future for our code. We started to write a few tests but we try to increase code coverage
- We implemented an automated API-Specification
- We will implement a feedback page as soon as the main features are working. This feature will be easily implemented due to our granular structured program
- We use the principles of clean-code-developement (KISS, DRY)
- We use widespread design principles for our frontend

### Other
- update class diagramm - [backend](https://github.com/dhbw-ka-tinf22b5-dinder/Dinder-SRS/blob/main/Diagramme/Klassendiagramme/Backend.png) (delete chat-class, divide service-class, optimize visualisation)

**Comment by ghost** [comment]
Das sieht wirklich sehr gut bei euch aus. Besonders gut finde ich den Punkt mit der API-Spezifikation, dass man in der Lage ist über url/api diese zu sehen. Eure Architektur Entscheidungen hören sich ebenfalls sehr gut an. Toll finde ich die Erwähnung von KISS und DRY. Meine einzige Anmerkung bezieht sich auf eure Tests. Ihr erwähnt ja TDD, also Test-Driven-Development, das Prinip zuerst die Tests und danach den Code zu schreiben (kurz gesagt). Jedoch erwähnt ihr danach, dass ihr erst jetzt nach der Implementierung anfangt die code coverage zu erhöhen. Bezieht sich TDD dann auf die Zukunkft, dass ihr TDD ab jetzt verwenden wollt?

**Comment by gloomyfrown** [comment]
Eure klare Fokussierung auf Interoperabilität, Modifizierbarkeit und Benutzerfreundlichkeit ist beeindruckend. Besonders positiv finde ich die geplante Implementierung von automatisierten Tests und die Anwendung von Clean-Code-Prinzipien. Das verspricht eine robuste und wartungsfreundliche Softwareentwicklung.
